### ISSUE TITLE
Feature Request: Detect presence of vehicle

### ISSUE THREAD

**Issue by murraykessell** [blakeblackshear/frigate#286]
I've notice that  the car detection only seems to work on car side profiles driving past.
I have a camera pointing down my driveway and it doesnt seem to detect a car pulling into the driveway and parking. This may be due to the lack of change in motion at slow speeds. Hence I was wondering if it was possible to include the ability to compare 2 images say X seconds apart and flag if something like a car has been detected in the second image ?

**Comment by blakeblackshear** [comment]
That may be a limitation of the model provided by Google. I don't understand your second question. Can you rephrase?

**Comment by murraykessell** [comment]
Thanks, re the second part. Rather than detect motion in an image, detect if an image taken now differs significantly from an image take say 5 seconds ago. Ie if there is no car in the first image of the driveway and then an image taken say 5 seconds later indicates significant change, or even detection of the presence of a car. I guess it could be similar to a frame rate of 0.2 fps. Is it possible to have sub 1 values for fps ? and does it work ?

**Comment by blakeblackshear** [comment]
I think what you are describing is how it already works. The motion detection runs for each frame by comparing the rolling average of previous frames. I tested a reduced frame rate for motion detection, and it didn't make a significant impact to CPU usage. If there is no motion, no detections are run.

**Comment by murraykessell** [comment]
I tried a frame rate of 0.2 but it doesn't detect any change when there is no car in the drive, and 5 seconds later there is. So I guess it's not "motion" per se but rather a change in the actual image.

**Comment by murraykessell** [comment]
Here's an example of what I mean.
![1](https://user-images.githubusercontent.com/51472089/98315774-fc0b5280-2013-11eb-8eb9-e54defca879a.jpg) ![2](https://user-images.githubusercontent.com/51472089/98315779-fd3c7f80-2013-11eb-8232-be869072fc28.jpg)
I can pull into the driveway and park slowly and I'm not detected (on a Pi 4 at 5fps). The second image varies greatly from the first. I would like to know when a change like this occurs in 2 camera images say 5 seconds apart and use it as a trigger. It's not really motion as you say, but it is a significant change.

**Comment by blakeblackshear** [comment]
I'm surprised frigate doesn't detect that already. Are you using a Coral?

**Comment by murraykessell** [comment]
No, do you think that is the problem ? In Australia they are a couple of hundred $ so I haven't bought one.

**Comment by blakeblackshear** [comment]
It is likely. Frigate discards frames if the processing pipeline is delayed so it can stay real-time. With really slow detection speeds, a lot of frames won't be analyzed at all.

**Comment by murraykessell** [comment]
OK thanks for your help, I'll put a Coral on the Christmas list and see if that fixes it
